lib/maths/CBoostedTreeFactory.cc

```diff
@@ -78,13 +88,13 @@ void CBoostedTreeFactory::initializeHyperparameterOptimisation() const {
     // less than p_1, this translates to using log parameter values.
 
     CBayesianOptimisation::TDoubleDoublePrVec boundingBox;
-    if (m_TreeImpl->m_LambdaOverride == boost::none) {
-        boundingBox.emplace_back(std::log(MIN_REGULARIZER_SCALE * m_TreeImpl->m_Lambda),
-                                 std::log(MAX_REGULARIZER_SCALE * m_TreeImpl->m_Lambda));
+    if (m_TreeImpl->m_RegularizationOverride.lambda() == boost::none) {
+        boundingBox.emplace_back(std::log(m_LambdaSearchInterval(MIN_REGULARIZER_INDEX)),
```

Comment:
No. In fact we arrange for it to be non-zero so the log doesn't explode. It can get so small that it is effectively negligible. For example, if it is much less than 1, for MSE loss, it doesn't really have any effect because we always have at least one row per leaf node.

(I have considered reparameterising so that we control total amount of regularisation and then distribute it out between different terms. This allows lambda to go to zero. However, that change also interacts with the upcoming regulariser based on tree depth. I plan to revisit this down-the-line.)